synapse/tests/test_lib_storm.py

```diff
@@ -3689,6 +3689,34 @@ async def pipecnt(self, runt, query, inq, outq):
                 self.len(4, nodes)
                 self.eq(4, tsks['cnt'])
 
+            self.len(20, await core.nodes('for $i in $lib.range(20) {[ test:str=$i ]}'))
+            q = '''
+            test:str
+            parallel --size 4 {
+                if (not $lib.vars.get(vals)) {
+                    $vals = ()
+                }
+                $vals.append($node.repr())
+                fini { $lib.print($lib.json.save($vals)) }
+            }
+            | spin
+            '''
+            vals = []
+            msgs = await core.stormlist(q)
+            for m in msgs:
+                if m[0] == 'print':
+                    vals.extend(s_json.loads(m[1]['mesg']))
```

Comment:
you could probably use `$lib.fire` to serialize the `$vals` and avoid needing to move them out of the runtime as json?

Reply:
updated